### ISSUE TITLE
Provide an option to remove validation attributes: schemaLocation & noNamespaceSchemaLocation

### ISSUE THREAD

**Issue by sdavids** [kristian/minify-xml#22]
```xml
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 https://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd">
  <url>
    <loc>https://test.com/index.html</loc>
    <lastmod>2023-10-28T18:30:00Z</lastmod>
  </url>
</urlset>
```

could be minified to:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
  <url>
    <loc>https://test.com/index.html</loc>
    <lastmod>2023-10-28T18:30:00Z</lastmod>
  </url>
</urlset>
```

**Comment by kristian** [comment]
Hey @sdavids, I am always happy to add new minifications. 👍 To be a valid minification (and not just removing content) is that the resulting document is semantically equivalent. Do you have any information of why the `schemaLocation` in your example is redundant and can be removed? Or in other words: What are the conditions when it is OK and not OK to remove the `schemaLocation` in the first place? Thanks.

**Comment by sdavids** [comment]
https://www.w3.org/TR/xmlschema-1/#xsi_schemaLocation

> The `xsi:schemaLocation` and `xsi:noNamespaceSchemaLocation` attributes can be used in a document to provide hints as to the physical location of schema documents which may be used for [·assessment·](https://www.w3.org/TR/xmlschema-1/#key-va).

https://www.w3.org/TR/xmlschema-1/#schema-loc

> `xsi:schemaLocation` and `xsi:noNamespaceSchemaLocation` [[attributes]](http://www.w3.org/TR/xml-infoset/#infoitem.element) can occur on any element.

https://www.w3.org/TR/xmlschema-1/#composition-schemaImport

> The [·actual value·](https://www.w3.org/TR/xmlschema-1/#key-vv) of the `schemaLocation`, if present, gives a hint as to where a serialization of a [·schema document·](https://www.w3.org/TR/xmlschema-1/#key-schemaDoc) with declarations and definitions for that namespace (or none) may be found. When no `schemaLocation` [[attribute]](http://www.w3.org/TR/xml-infoset/#infoitem.element) is present, the schema author is leaving the identification of that schema to the instance, application or user, via the mechanisms described below in [Layer 3: Schema Document Access and Web-interoperability (§4.3)](https://www.w3.org/TR/xmlschema-1/#composition-instances).

> **Note**: Since both the namespace and schemaLocation [[attribute]](http://www.w3.org/TR/xml-infoset/#infoitem.element) are optional […]

⇒ In short, if one does not care about validation, one can remove the `schemaLocation` and `noNamespaceSchemaLocation` attributes; also, if after that removal no tag/attribute uses the `xsi` namespace it can be removed as well.

---


The sitemap example provided on the official site does not use `schemaLocation`:

https://www.sitemaps.org/protocol.html#mainContent

```xml
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
   <url>
      <loc>http://www.example.com/</loc>
      <lastmod>2005-01-01</lastmod>
      <changefreq>monthly</changefreq>
      <priority>0.8</priority>
   </url>
</urlset> 
```

---

While authoring a sitemap a `schemaLocation` is useful for auto-completion and validation though.

But the version uploaded to one's server does not need them.

I doubt that consumers of sitemaps (search engines) use validating parsers due to the memory, performance, and security implications.

**Comment by kristian** [comment]
Ok, I see your point. I especially like the idea of also removing the `xsi` namespace in general, if the `schemaLocation` was removed and was the only reference to `xsi`. I think it makes sense to add that as an option, as you suggested, so if one would want to keep the option to validate the schema, they could still keep the attributes in. Let me see what I can do.

**Comment by sdavids** [comment]
Thanks,

unfortunately they can appear everywhere and not only at the root 🫤

**Comment by kristian** [comment]
Added with c88d0cc68d17f9a0a6ff21401d421c0e60c201bf and released version `4.2.0` to NPM. Hope this helps @sdavids.

**Comment by sdavids** [comment]
I verified my use case with `4.2.0`, thanks!
